src/core/util/props.js

```diff
@@ -122,13 +122,14 @@ function assertProp (
       type = [type]
     }
     for (let i = 0; i < type.length && !valid; i++) {
-      const assertedType = assertType(value, type[i])
+      const assertedType = assertType(value, type[i], vm)
       expectedTypes.push(assertedType.expectedType || '')
       valid = assertedType.valid
     }
   }
 
-  if (!valid) {
+  const haveExpectedTypes = expectedTypes.filter(t => t).length;
```

Comment:
```suggestion
  const haveExpectedTypes = expectedTypes.some(t => t)
```
Also, remove the semicolons to adhere to the codestyle?